### PR TITLE
Vcluster/2361 shared nuget for setting up EnergyOrigin solutions

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

File: libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/Swagger/ConfigureSwaggerOptions.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    },
+                    new string[] { }
+                }
+            });
+
+        foreach (var description in provider.ApiVersionDescriptions)
+        {
+            options.SwaggerDoc(description.GroupName, new OpenApiInfo
+            {
+                Title = $"{subsystemName.FirstCharToUpper()} Query API {description.ApiVersion}",
```
Comment (ckr123) [inline libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/Swagger/ConfigureSwaggerOptions.cs:53]: Could this title be a variable?

File: libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/EnergyOrigin.Setup.csproj [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	</PropertyGroup>
+
+	<ItemGroup>
+		<FrameworkReference Include="Microsoft.AspNetCore.App" />
+	</ItemGroup>
+
+	<ItemGroup>
+	  <PackageReference Include="Asp.Versioning.Mvc" Version="8.0.0" />
+	  <PackageReference Include="Asp.Versioning.Mvc.ApiExplorer" Version="8.0.0" />
+	  <PackageReference Include="MassTransit" Version="8.1.3" />
```
Comment (ckr123) [inline libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/EnergyOrigin.Setup.csproj:23]: I dont believe all our systems are using masstransit. Do we still want it here or could we make some kind of builder pattern on the metrics/traces so it is optional

**Review by martinhenningjensen — APPROVED** [review]

File: libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/Swagger/ConfigureSwaggerOptions.cs (hunk 1 above)
Comment (tnickelsen) [reply libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/Swagger/ConfigureSwaggerOptions.cs]: removed query from the string

File: libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/EnergyOrigin.Setup.csproj (hunk 2 above)
Comment (tnickelsen) [reply libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/EnergyOrigin.Setup.csproj:23]: We keep it to have masstransit tracing on on default. They will just be empty if there's no masstransit in the project

File: libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/IServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            .AddApiExplorer();
+    }
+
+    public static void AddControllersWithEnumsAsStrings(this IServiceCollection services)
+    {
+        services.AddControllers()
+            .AddJsonOptions(o =>
+                o.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter()));
+    }
+}
```
Comment (sahma19) [inline libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/IServiceCollectionExtensions.cs:37]: If this is added to the NuGet, are we then forcing everyone who imports it, to use versioning in our api?

I suggest having a boolean as a featureflag, that makes one able toggle on/off, each feature that is included in the NuGet. Be it MassTransit, Swagger, JsonEnumStringConverter, etc.

**Review by sahma19 — CHANGES_REQUESTED** [review]

File: libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/IServiceCollectionExtensions.cs (hunk 3 above)
Comment (tnickelsen) [reply libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/IServiceCollectionExtensions.cs:37]: No only if you use the specified function for versioning. I would rather make a new function for without versioning, if that is needed, than making a boolean. But do we have apis that does not need versioning?

**Review by ckr123 — APPROVED** [review]

**Review by sahma19 — APPROVED** [review]

**Pull request merged**